packages/api/tests/testEnvironment.js

```diff
@@ -115,6 +115,8 @@ async function tableCleanup(knex) {
     DELETE FROM "pesticide";
     DELETE FROM "task_type";
     DELETE FROM "farmDataSchedule";
+    DELETE FROM "default_animal_breed";
```

Comment:
So I just ran into an Error created from #3078 . Because the defaults are deleted from the table at the end the testing db migration fails when referencing the types. 

I think either we do not want to be deleting the defaults after the test is complete or we want to populate the db afterward similar to (but not beforeAll) populateTaskTypes. 

![Screenshot 2024-01-22 at 11 55 39 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/133dca4f-4d8e-4216-ad71-56084bbe2514)

To reproduce this would be Run the tests after running LF-3972b migration. Then try to run this migration on testing db after the table has been cleaned up.

Reply:
The way I structured the tests I actually don't want any defaults populated since the tests themselves take care of populating default data as they need, so I think I'll just adjust the migration so that it won't insert the breed if it doesn't find the type

Reply:
So yeah I don't really mind what ever you want to do in tests. I just don't think I like the idea of skipping if it doesn't exist. It is a useful error if for some reason you misspelled the custom types here compared to Joyces.  I would want to see a failed migration.

I don't see my scenario as one off but a consequence of testing in order. I am a little surprised more of us didn't run into it. So personally if it is between leaving it as is and having someone like me who tested it in order rollback their db or adding the special case -- I think I would prefer to leave it as is and I just rollback my db and not run tests again on each branch until it is all merged into integration as one.